internal/impl/gcp/cloud_storage_output.go

```diff
@@ -191,17 +196,56 @@ func (g *gcpCloudStorageOutput) WriteWithContext(ctx context.Context, msg types.
 			return nil
 		})
 
-		w := client.Bucket(g.conf.Bucket).Object(g.path.String(i, msg)).NewWriter(ctx)
+		outputPath := g.path.String(i, msg)
+		_, err := client.Bucket(g.conf.Bucket).Object(outputPath).Attrs(ctx)
```

Comment:
Is this needed? It looks like a network call but I'm not sure what it's for.

Reply:
Yea, that's needed to check if the object at `outputPath` exists. The attributes object isn't needed from that call, just need to see if `err` is the ErrObjectNotExist error.

Reply:
Ah okay, in that case I think we ought to skip it when we're set to overwrite on collision as it's an unused network call in those circumstances.